pkg/action/bootentries.go

```diff
@@ -0,0 +1,212 @@
+	if err != nil {
+		return fmt.Errorf("could not read host partitions: %s", err)
+	}
+	for _, p := range parts {
+		if p.FilesystemLabel == constants.EfiLabel {
+			efiPartition = p
+			break
+		}
+	}
+	if efiPartition == nil {
+		return fmt.Errorf("could not find EFI partition")
+	}
+
+	// Mount it RW
+	err = syscall.Mount("", efiPartition.MountPoint, "", syscall.MS_REMOUNT, "")
+	if err != nil {
+		cfg.Logger.Errorf("could not remount EFI partition: %s", err)
+		return err
+	}
+	// Remount it RO when finished
+	defer func(source string, target string, fstype string, flags uintptr, data string) {
+		err = syscall.Mount(source, target, fstype, flags, data)
+		if err != nil {
+			cfg.Logger.Errorf("could not remount EFI partition as RO: %s", err)
+		}
+	}("", efiPartition.MountPoint, "", syscall.MS_REMOUNT|syscall.MS_RDONLY, "")
+
+	systemdConf, err := SystemdBootConfReader(filepath.Join(efiPartition.MountPoint, "loader/loader.conf"))
+	if err != nil {
+		cfg.Logger.Errorf("could not read loader.conf: %s", err)
+		return err
+	}
+	// Set the default entry to the selected entry
+	systemdConf["default"] = entry
+	err = SystemdBootConfWriter(filepath.Join(efiPartition.MountPoint, "loader/loader.conf"), systemdConf)
+	if err != nil {
+		cfg.Logger.Errorf("could not write loader.conf: %s", err)
+		return err
+	}
+	cfg.Logger.Infof("Default boot entry set to %s", entry)
+	return err
+}
+
+func ListBootEntries(cfg *config.Config) error {
+	// Get EFI partition
+	efiPartition, err := GetEfiPartition()
+	if err != nil {
+		return err
+	}
+	// Get default entry from loader.conf
+	loaderConf, err := utils.SystemdBootConfReader(filepath.Join(efiPartition.MountPoint, "loader/loader.conf"))
+	if err != nil {
+		return err
+	}
+
+	entries, err := ListSystemdEntries(cfg, efiPartition)
+
+	// create a selector
+	selector := selection.New(fmt.Sprintf("Select Boot Entry (current entry: %s)", loaderConf["default"]), entries)
+	selector.Filter = nil        // Remove the filter
+	selector.ResultTemplate = `` // Do not print the result as we are asking for confirmation afterwards
+	selected, _ := selector.RunPrompt()
+	c := confirmation.New("Are you sure you want to change the boot entry to "+selected, confirmation.Yes)
+	c.ResultTemplate = ``
+	confirm, err := c.RunPrompt()
+	if confirm {
+		return SelectBootEntry(cfg, selected)
+	}
+	return err
+}
+
+// SystemdBootConfReader reads a systemd-boot conf file and returns a map with the key/value pairs
+func SystemdBootConfReader(filePath string) (map[string]string, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer func(file *os.File) {
+		_ = file.Close()
+	}(file)
+
+	result := make(map[string]string)
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		parts := strings.SplitN(line, " ", 2)
+		if len(parts) == 2 {
+			result[parts[0]] = parts[1]
+		}
+		if len(parts) == 1 {
+			result[parts[0]] = ""
+		}
+	}
```

Comment:
if I remember correctly on the other story we were pairing with @jimmykarily we were using something similar, maybe it's already time to introduce this as an abstraction for both codes to share

Reply:
Yes, this was originally here I think and moved into the ask but not merged yet and I didn't want to block this or force Dimitri's patch to get it, so this is here as a workaround to be able to test this and see the PR, but this functions belong in ~~a museum~~ the sdk

Reply:
good point, the refactoring can come later. I created a ticket https://github.com/kairos-io/kairos/issues/2252